packages/commonwealth/client/scripts/state/api/comments/deleteComment.ts

```diff
@@ -0,0 +1,95 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import axios from 'axios';
+import app from 'state';
+import { ApiEndpoints } from 'state/api/config';
+import { updateThreadInAllCaches } from '../threads/helpers/cache';
+import useFetchCommentsQuery from './fetchComments';
+
+interface DeleteCommentProps {
+  address: string;
+  chainId: string
+  canvasHash: string
+  commentId: number;
+  existingNumberOfComments: number;
+}
+
+const deleteComment = async ({
+  address,
+  chainId,
+  commentId,
+  canvasHash,
+}: DeleteCommentProps) => {
+  const {
+    session = null,
+    action = null,
+    hash = null,
+  } = await app.sessions.signDeleteComment({
+    comment_id: canvasHash,
+  })
+
+  await axios.delete(`${app.serverUrl()}/comments/${commentId}`, {
+    data: {
+      jwt: app.user.jwt,
+      address: address,
+      chain: chainId,
+      author_chain: chainId
+    },
+  })
+
+  // Important: we render comments in a tree, if this deleted comment was
+  // the root comment of a tree, then we want to preserve the comment tree,
+  // but in place of this deleted comment we will show the "[deleted]" msg.
```

Comment:
is this actually something that was discussed with the product team?

Reply:
This is actually a behavior in production. I just added explainer comments after bumping into issues (learning the hard way lol 😅)

Reply:
thats better @jnaviask, will update that

Reply:
Done